src/main/java/httpserver/HttpServer.java

```diff
@@ -0,0 +1,48 @@
+package httpserver;
+
+import httpApplication.Application;
+import server.ConnectionDataStream;
+import server.Process;
+
+import java.io.IOException;
+import java.util.HashMap;
+
+public class HttpServer implements Process {
+    public HttpServer(Application application) {
+        this.application = application;
+    }
+
+    public void runWith(ConnectionDataStream client) {
+
+        try {
+            HttpIncomingMessage clientRequest = new HttpRequest(parseRawRequest(client.readInput()));
+            HttpResponse applicationResponse = application.start(clientRequest);
+
+            client.write(convertHttpResponseToRawRequest(applicationResponse));
+            client.close();
```

Comment:
What happens if there is an exception? Will the client get closed? Or will this leak?

Reply:
I added a log, so an I/O error occurred messaged is sent to the host server log.